src/bin/bat/app.rs

```diff
@@ -198,22 +198,23 @@ impl App {
                     }
                 })
                 .unwrap_or_else(|| String::from(HighlightingAssets::default_theme())),
-            visible_lines: if self.matches.is_present("diff") {
-                VisibleLines::DiffContext(
+            visible_lines: match self.matches.is_present("diff") {
+                #[cfg(feature = "git")]
+                true => VisibleLines::DiffContext(
```

Comment:
I see why you chose to use a `match` statement here, but this is a bit difficult to read. I don't know a better solution of the top of my head, though.

I guess `if cfg!(…)` does not work because `DiffContext` is not available if `git` is not set?

Reply:
> I guess if cfg!(…) does not work because DiffContext is not available if git is not set?
Pretty much. It's a bit difficult to read, but it was the smallest change to make it work like so:

1. If `--diff` is present with git support, use the diff context.
2. Otherwise, use the `--line-range` option.

Reply:
Ok